scripts/ci/run_cli_tool.sh

```diff
@@ -0,0 +1,140 @@
+        echo "chmod +x /usr/bin/${NAME} /usr/bin/${NAME}-update"
+    done
+    exit 0
+fi
+
+COMMON_DOCKER_ARGS=(
+    -v "${HOST_AIRFLOW_SOURCES}/tmp:/tmp"
+    -v "${HOST_AIRFLOW_SOURCES}/files:/files"
+    -v "${HOST_AIRFLOW_SOURCES}:/opt/airflow"
+    --env-file <(env)
+)
+
+AWS_CREDENTIALS_DOCKER_ARGS=(-v "${HOST_HOME}/.aws:/root/.aws")
+AZURE_CREDENTIALS_DOCKER_ARGS=(-v "${HOST_HOME}/.azure:/root/.azure")
+GOOGLE_CREDENTIALS_DOCKER_ARGS=(-v "${HOST_HOME}/.config/gcloud:/root/.config/gcloud")
+
+DIRECTORIES_TO_FIX=('/tmp/' '/files/')
+
+COMMAND=("${@}")
+
+# Configure selected tool
+case "${TOOL_NAME}" in
+    aws )
+        COMMON_DOCKER_ARGS+=("${AWS_CREDENTIALS_DOCKER_ARGS[@]}")
+        DIRECTORIES_TO_FIX+=("${HOST_HOME}/.aws")
+        IMAGE_NAME="amazon/aws-cli:latest"
+        ;;
+    az )
+        COMMON_DOCKER_ARGS+=("${AZURE_CREDENTIALS_DOCKER_ARGS[@]}")
+        DIRECTORIES_TO_FIX+=("${HOST_HOME}/.azure")
+        IMAGE_NAME="mcr.microsoft.com/azure-cli:latest"
+        ;;
+    gcloud | bq | gsutil )
+        COMMON_DOCKER_ARGS+=("${GOOGLE_CREDENTIALS_DOCKER_ARGS[@]}")
+        DIRECTORIES_TO_FIX+=("${HOST_HOME}/.config/gcloud")
+        IMAGE_NAME="gcr.io/google.com/cloudsdktool/cloud-sdk:latest"
+        COMMAND=("$TOOL_NAME" "${@}")
+        ;;
+    terraform )
+        COMMON_DOCKER_ARGS+=("${GOOGLE_CREDENTIALS_DOCKER_ARGS[@]}")
+        COMMON_DOCKER_ARGS+=("${AZURE_CREDENTIALS_DOCKER_ARGS[@]}")
+        COMMON_DOCKER_ARGS+=("${AWS_CREDENTIALS_DOCKER_ARGS[@]}")
+        COMMON_DOCKER_ARGS+=(-w /opt/airflow)
+        DIRECTORIES_TO_FIX+=(
+            "${HOST_HOME}/.config/gcloud"
+            "${HOST_HOME}/.aws"
+            "${HOST_HOME}/.azure"
+        )
+        IMAGE_NAME="hashicorp/terraform:latest"
+        ;;
+    java )
+        # TODO: Should we add other credentials?
+        COMMON_DOCKER_ARGS+=("${GOOGLE_CREDENTIALS_DOCKER_ARGS[@]}")
+        DIRECTORIES_TO_FIX+=("${HOST_HOME}/.config/gcloud")
+        IMAGE_NAME="openjdk:8-jre-slim"
+        COMMAND=("/usr/local/openjdk-8/bin/java" "${@}")
+        ;;
+    * )
+        >&2 echo "Unsupported tool name: ${TOOL_NAME}"
+        exit 1
+        ;;
+esac
+
+# Run update, if requested
+if [[ "${SCRIPT_NAME}" == *-update ]]; then
+    docker pull "${IMAGE_NAME}"
+    exit $?
+fi
+
+# Otherwise, run tool
+TOOL_DOCKER_ARGS=(--rm --interactive)
+TOOL_DOCKER_ARGS+=("${COMMON_DOCKER_ARGS[@]}")
+
+if [ -t 0 ] ; then
+    TOOL_DOCKER_ARGS+=(
+        --tty
+    )
+fi
+
+docker run "${TOOL_DOCKER_ARGS[@]}" "${IMAGE_NAME}" "${COMMAND[@]}"
+
+RES=$?
+
+# Set file permissions to the host user
+if [[ "${HOST_OS}" == "Linux" ]]; then
+    FIX_DOCKER_ARGS=(--rm)
+    FIX_DOCKER_ARGS+=("${COMMON_DOCKER_ARGS[@]}")
+    FIX_COMMAND=(bash -c
+        "find ${DIRECTORIES_TO_FIX[@]@Q} -user root -print0 | xargs --null chown '${HOST_USER_ID}.${HOST_GROUP_ID}' --no-dereference")
+
+    docker run "${FIX_DOCKER_ARGS[@]}" "${AIRFLOW_CI_IMAGE}" "${FIX_COMMAND[@]}" >/dev/null 2>&1
```

Comment:
@potiuk Why we need docker here? I don't think that is necessary.

Reply:
Because those files are owned by root  and this way we get root access to those files and can change the ownership. Otherwise you need to use sudo and provide password.  This is much more convenient.

Reply:
But we run this script in the container. Do we have to run the container in the container? I think breeze uses root by default, so it should work. This is a problem if the script would be run by the host but it is not. This script is run in the container.

```
root@09797f7fa137:/opt/airflow# id
uid=0(root) gid=0(root) groups=0(root)
```

Reply:
ah yeah . Silly me.  indeed we run it from the container :)

Reply:
> > Why /root/ here ? The volumes are mapped from host not from the container because we are using the same engine that is in the host
> 
> We use these paths to determine which directories to fix, but we do it in the container, so we need to use the path from the container. Saying differently, these paths are used in scripts that run in the container.

Once we run it no via another container (I need some rest) - then yes

Reply:
But ..... it makes sense, because we mount additional volumes with credentials. The main container cannot access them. No container should have access to them unless ``--forward-credentials`` flag is passed, but that's another problem. I want to take care of them later.

Reply:
Ah.Ok. Now I remember why I did that indeed (tired again). I wanted to make sure that this works without --forward-credentials at all.. 

There is no reason why we should not forward credentials to the official gcloud/ aws/ etc. tools . Forwarding credentials just to those tools is equivalent to running them locally, so there is no risk. So indeed both running docker and using HOME made sense from the start.